src/cli/cli.cpp

```diff
@@ -276,11 +276,34 @@ template <> otError Interpreter::Process<Cmd("version")>(Arg aArgs[])
 
 template <> otError Interpreter::Process<Cmd("reset")>(Arg aArgs[])
 {
-    OT_UNUSED_VARIABLE(aArgs);
+    otError error = OT_ERROR_NONE;
 
-    otInstanceReset(GetInstancePtr());
+    if (aArgs[0].IsEmpty())
+    {
+        otInstanceReset(GetInstancePtr());
+    }
 
-    return OT_ERROR_NONE;
+#if OPENTHREAD_CONFIG_PLATFORM_BOOTLOADER_MODE_ENABLE
+    /**
+     * @cli reset bootloader
+     * @code
+     * reset bootloader
+     * @endcode
```

Comment:
```suggestion
     * @code
     * reset bootloader
     * Done
     * @endcode
```

Reply:
Sorry, forgot to mention we need Done output at the end of the example

Reply:
@Vyrastas The `reset bootloader` command doesn't print `Done`, similar to how `factoryreset` and `reset` don't print `Done`

Reply:
Ok cool, didn't know that.  All good then